### PR TITLE
Drop Python < 3.8 and replace pkg_resources with importlib

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Review by evgeni — COMMENTED** [review]
This would kill Python 2 compatibility, no?

**Comment by evgeni** [comment]
Thanks! Looks good then.

One oddity I still can offer. https://github.com/theforeman/jenkins-jobs/blob/master/theforeman.org/pipelines/lib/obal.groovy does not `pip install` anything, but just checks out obsah and obal and mangles `PYTHONPATH`. That means it'll break with this change unless we somehow ensure importlib things are installed.

(`pip install` probably won't do as we don't use a `venv` right now? need to investigate)

File: setup.py [hunk 1]
```diff
@@ -68,6 +68,8 @@ def find_package_data(package, data_dir):
     install_requires=[
         'ansible >= 2.5; python_version < "3.8"',
         'ansible-core; python_version >= "3.8"',
+        'importlib_resources; python_version < "3.7"',
+        'importlib_metadata; python_version < "3.8"',
```
Comment (evgeni) [inline setup.py:72]: given on EL8 we use py3.11 now, this is safe to merge once we turn of the el7 builders.

Comment (ekohl) [reply setup.py:72]: Should we then refactor this in a "drop Python < 3.8 compatibility" PR?

Comment (evgeni) [reply setup.py]: I think downstream it's still py3.6 (with whatever ansible is compatible with that)

Comment (ekohl) [reply setup.py]: Do we still need to care about this or can we assume Python 3.8+ by now?

**Review by evgeni — APPROVED** [review]

**Pull request merged**